src/control/ros/Conversions.cpp

```diff
@@ -0,0 +1,288 @@
+
+  assert(_currPosition.size() == numDofs);
+  assert(_nextPosition.size() == numDofs);
+  splineProblem.addConstantConstraint(0, 0, _currPosition);
+  splineProblem.addConstantConstraint(1, 0, _nextPosition);
+
+  if (_numCoefficients >= 4)
+  {
+    assert(_currVelocity.size() == numDofs);
+    assert(_nextVelocity.size() == numDofs);
+    splineProblem.addConstantConstraint(0, 1, _currVelocity);
+    splineProblem.addConstantConstraint(1, 1, _nextVelocity);
+  }
+
+  if (_numCoefficients >= 6)
+  {
+    assert(_currAcceleration.size() == numDofs);
+    assert(_nextAcceleration.size() == numDofs);
+    splineProblem.addConstantConstraint(0, 2, _currAcceleration);
+    splineProblem.addConstantConstraint(1, 2, _nextAcceleration);
+  }
+
+  const auto splineSegment = splineProblem.fit();
+  return splineSegment.getCoefficients()[0];
+}
+
+//=============================================================================
+void extractJointTrajectoryPoint(
+  const trajectory_msgs::JointTrajectory& _trajectory,
+  size_t _index, size_t _numDofs,
+  Eigen::VectorXd* _positions, bool _positionsRequired,
+  Eigen::VectorXd* _velocities, bool _velocitiesRequired,
+  Eigen::VectorXd* _accelerations, bool _accelerationsRequired,
+  std::map<size_t, size_t> indexMap)
+{
+  const auto& waypoint = _trajectory.points[_index];
+
+  try
+  {
+    Eigen::VectorXd trajPos, trajVel, trajAccel;
+    checkVector("positions", waypoint.positions, _numDofs,
+      _positionsRequired, &trajPos);
+    checkVector("velocities", waypoint.velocities, _numDofs,
+      _velocitiesRequired, &trajVel);
+    checkVector("accelerations", waypoint.accelerations, _numDofs,
+      _accelerationsRequired, &trajAccel);
+
+    if ( trajPos.size() > 0 )
+      reorder(indexMap, trajPos, _positions);
+    if ( trajVel.size() > 0 )
+      reorder(indexMap, trajVel, _velocities);
+    if ( trajAccel.size() > 0 )
+      reorder(indexMap, trajAccel, _accelerations);
+  }
+  catch (const std::invalid_argument& e)
+  {
+    std::stringstream message;
+    message << "Waypoint " << _index << " is invalid: " << e.what();
+    throw std::invalid_argument(message.str());
+  }
+}
+
+//=============================================================================
+// The rows of inVector is reordered in outVector.
+void reorder(std::map<size_t, size_t> indexMap,
+  const Eigen::VectorXd& inVector, Eigen::VectorXd* outVector)
+{
+  *outVector = Eigen::VectorXd::Zero(inVector.size());
+  for (auto it = indexMap.begin(); it != indexMap.end(); ++it)
+  {
+    (*outVector)(it->second) = inVector(it->first);
+  }
+}
+
+//=============================================================================
+dart::dynamics::Joint* findJointByName(
+    dart::dynamics::MetaSkeleton* metaSkeleton, const std::string& jointName)
+{
+  for (size_t i = 0; i < metaSkeleton->getNumJoints(); ++i)
+  {
+    if (metaSkeleton->getJoint(i)->getName() == jointName)
+      return metaSkeleton->getJoint(i);
+  }
```

Comment:
@jslee02 That would be a great addition. 👍

In fact, it would be nice to have two functions: (1) one that returns a list of joints and (2) one that returns a single joint and prints a warning if multiple joints have the same name.